client/shipping-label/state/actions.js

```diff
@@ -514,13 +515,18 @@ export const purchaseLabel = () => ( dispatch, getState, context ) => {
 					labelId: label.label_id,
 				} ) );
 				const state = getState().shippingLabel;
-				printDocument( getPrintURL( state.paperSize, labelsToPrint, context ) )
-					.then( () => {
+				const printUrl = getPrintURL( state.paperSize, labelsToPrint, context );
+				printDocument( printUrl, false )
+					.then( ( success ) => {
 						const noticeText = 1 === response.length
 								? __( 'Your shipping label was purchased successfully' )
 								: sprintf( __( 'Your %d shipping labels were purchased successfully' ), response.length );
 						dispatch( NoticeActions.successNotice( noticeText ) );
-						dispatch( exitPrintingFlow( true ) );
+						if ( success ) {
+							dispatch( exitPrintingFlow( true ) );
+						} else {
+							dispatch( { type: SHOW_PRINT_CONFIRMATION, printUrl } );
```

Comment:
We still need a way to indicate the purchase dialog "Purchase has completed, but don't close yourself yet". That has to go into the redux state tree anyway, so at least I prefer to make it explicit here.